components/edit-collective/EditGoals.js

```diff
@@ -100,32 +100,30 @@ class EditGoals extends React.Component {
       const updatedGoal = { ...goal, type: goal.type || this.defaultType, [fieldName]: value };
       const updatedGoals = [...state.goals];
       updatedGoals[index] = updatedGoal;
-      return { ...state, isTouched: true, goals: updatedGoals };
+      return { isTouched: true, goals: updatedGoals };
     });
   };
 
   toggleGoalsOnCollectivePage = ({ checked }) => {
     this.setState(state => ({
-      ...state,
       isTouched: true,
       collectivePage: { ...state.collectivePage, showGoals: checked },
     }));
   };
 
   addGoal = () => {
     const newGoal = { type: this.defaultType, key: uuid() };
-    this.setState(state => ({ ...state, isTouched: true, goals: [...state.goals, newGoal] }));
+    this.setState(state => ({ isTouched: true, goals: [...state.goals, newGoal] }));
   };
 
   removeGoal = index => {
     this.setState(state => {
-      if (index < 0 || index > state.goals.length) {
-        return state;
-      } else {
+      if (index >= 0 && index <= state.goals.length) {
```

Comment:
That is my last feedback, I don't see any reason to change this test, so let's revert it and return null.